packages/node_modules/cerebral/src/views/preact/preact.test.js

```diff
@@ -0,0 +1,380 @@
+      assert.equal(renderCount, 1)
+      controller.getSignal('test')()
+      assert.equal(renderCount, 2)
+      assert.equal($('div').innerHTML, 'bar2')
+    })
+    it('should rerender affected components', () => {
+      let renderCount = 0
+      const controller = Controller({
+        state: {
+          foo: 'bar',
+        },
+        signals: {
+          methodCalled: [({ state }) => state.set('foo', 'bar2')],
+        },
+      })
+      class TestComponentClass2 extends Component {
+        render() {
+          renderCount++
+          return <div />
+        }
+      }
+      const TestComponent2 = connect({}, TestComponentClass2)
+      class TestComponentClass extends Component {
+        render() {
+          renderCount++
+          return <div><TestComponent2 /></div>
+        }
+      }
+      const TestComponent = connect(
+        {
+          foo: state`foo`,
+        },
+        TestComponentClass
+      )
+      mount(
+        <Container controller={controller}>
+          <TestComponent />
+        </Container>
+      )
+      controller.getSignal('methodCalled')()
+      assert.equal(renderCount, 3)
+    })
+    it('should allow props tag and computed', () => {
+      const controller = Controller({
+        state: {
+          foo: {
+            baz: 'mip',
+          },
+        },
+      })
+      const TestComponent = connect(
+        {
+          a: compute(props`foo`),
+          b: props`bar.id`,
+          c: state`foo.${props`propKey`}`,
+        },
+        ({ a, b, c }) => {
+          return <div>{a + b + c}</div>
+        }
+      )
+      mount(
+        <Container controller={controller}>
+          <TestComponent foo="bar" bar={{ id: '1' }} propKey="baz" />
+        </Container>
+      )
+      assert.equal($('div').innerHTML, 'bar1mip')
+    })
+    it('should update on props change', done => {
+      const controller = Controller({})
+      class TestComponentClass2 extends Component {
+        render() {
+          return <div>{this.props.foo}</div>
+        }
+      }
+      const TestComponent2 = connect({}, TestComponentClass2)
+      class TestComponentClass extends Component {
+        constructor(props) {
+          super(props)
+          this.state = { foo: 'bar' }
+        }
+        changePath() {
+          this.setState({
+            foo: 'bar2',
+          })
+        }
+        render() {
+          return <TestComponent2 foo={this.state.foo} />
+        }
+      }
+      const TestComponent = connect({}, TestComponentClass)
+      const tree = mount(
+        <Container controller={controller}>
+          <TestComponent />
+        </Container>
+      )
+      assert.equal(
+        tree._component._component._component instanceof TestComponentClass,
+        true
+      )
+      assert.equal($('div').innerHTML, 'bar')
+      const component = tree._component._component._component
+      component.changePath()
+      setTimeout(() => {
```

Comment:
Try rather passing a callback to `changePath`, and then use that callback on `setState({}, cb}`... the rendering might be async :-)